src/Algebra/Lattice/Bundles.agda

```diff
@@ -0,0 +1,184 @@
+    _∧_               : Op₂ Carrier
+    isMeetSemilattice : IsSemilattice _≈_ _∧_
+
+  open IsMeetSemilattice _≈_ isMeetSemilattice public
+
+  semilattice : Semilattice c ℓ
+  semilattice = record { isSemilattice = isMeetSemilattice }
+
+  open Semilattice semilattice public using (rawMagma; magma; semigroup; band)
+
+record JoinSemilattice c ℓ : Set (suc (c ⊔ ℓ)) where
+  infixr 7 _∨_
+  infix  4 _≈_
+  field
+    Carrier           : Set c
+    _≈_               : Rel Carrier ℓ
+    _∨_               : Op₂ Carrier
+    isJoinSemilattice : IsSemilattice _≈_ _∨_
+
+  open IsJoinSemilattice _≈_ isJoinSemilattice public
+
+  semilattice : Semilattice c ℓ
+  semilattice = record { isSemilattice = isJoinSemilattice }
+
+  open Semilattice semilattice public using (rawMagma; magma; semigroup; band)
+
+
+record BoundedSemilattice c ℓ : Set (suc (c ⊔ ℓ)) where
+  infixr 7 _∙_
+  infix  4 _≈_
+  field
+    Carrier              : Set c
+    _≈_                  : Rel Carrier ℓ
+    _∙_                  : Op₂ Carrier
+    ε                    : Carrier
+    isBoundedSemilattice : IsBoundedSemilattice _≈_ _∙_ ε
+
+  open IsBoundedSemilattice _≈_ isBoundedSemilattice public
+
+  semilattice : Semilattice c ℓ
+  semilattice = record { isSemilattice = isSemilattice }
+
+  open Semilattice semilattice public using (rawMagma; magma; semigroup; band)
+
+record BoundedMeetSemilattice c ℓ : Set (suc (c ⊔ ℓ)) where
+  infixr 7 _∧_
+  infix  4 _≈_
+  field
+    Carrier                  : Set c
+    _≈_                      : Rel Carrier ℓ
+    _∧_                      : Op₂ Carrier
+    ⊤                        : Carrier
+    isBoundedMeetSemilattice : IsBoundedSemilattice _≈_ _∧_ ⊤
+
+  open IsBoundedMeetSemilattice _≈_ isBoundedMeetSemilattice public
+
+  boundedSemilattice : BoundedSemilattice c ℓ
+  boundedSemilattice = record
+    { isBoundedSemilattice = isBoundedMeetSemilattice }
+
+  open BoundedSemilattice boundedSemilattice public
+    using (rawMagma; magma; semigroup; band; semilattice)
+
+record BoundedJoinSemilattice c ℓ : Set (suc (c ⊔ ℓ)) where
+  infixr 7 _∨_
+  infix  4 _≈_
+  field
+    Carrier                  : Set c
+    _≈_                      : Rel Carrier ℓ
+    _∨_                      : Op₂ Carrier
+    ⊥                        : Carrier
+    isBoundedJoinSemilattice : IsBoundedSemilattice _≈_ _∨_ ⊥
+
+  open IsBoundedJoinSemilattice _≈_ isBoundedJoinSemilattice public
+
+  boundedSemilattice : BoundedSemilattice c ℓ
+  boundedSemilattice = record
+    { isBoundedSemilattice = isBoundedJoinSemilattice }
+
+  open BoundedSemilattice boundedSemilattice public
+    using (rawMagma; magma; semigroup; band; semilattice)
+
+
+record Lattice c ℓ : Set (suc (c ⊔ ℓ)) where
+  infixr 7 _∧_
+  infixr 6 _∨_
+  infix  4 _≈_
+  field
+    Carrier   : Set c
+    _≈_       : Rel Carrier ℓ
+    _∨_       : Op₂ Carrier
+    _∧_       : Op₂ Carrier
+    isLattice : IsLattice _≈_ _∨_ _∧_
+
+  open IsLattice isLattice public
```

Comment:
Should you also not be exporting (and renaming to) things like `∧-semilattice` here? Ditto for the two modules below as well.